Tests/Form/Type/SimpleMoneyTypeTest.php

```diff
@@ -72,12 +62,33 @@ public function testGreaterThan1000Valid()
     public function testSetData()
     {
         \Locale::setDefault("fr_FR");
-        $moneyType = new SimpleMoneyType($this->pairManager, 2);
-        $form = $this->factory->create($moneyType, null, array());
+        $form = $this->factory->create($this->simpleMoneyTypeClass, null, array());
         $form->setData(Money::EUR(120));
         $formView = $form->createView();
 
         $this->assertEquals("1,20", $formView->children["tbbc_amount"]->vars["value"]);
     }
 
+    protected function getExtensions()
+    {
+        //This is probably not ideal, but I'm not sure how to set up the pair manager
+        // with different decimals for different tests in Symfony 3.0
```

Comment:
@drewclauson I did not mean to say that options would go to constructor. Constructor should have some default settings (as nulls[?]) which would then be set as defaults via `$resolver->setDefaults()` and which could be changed when you create it with form factory with third param as shown above:

```
$form = $this->factory->create('Tbbc\MoneyBundle\Form\Type\CurrencyType', null, array(
    'reference_currency' => 'EUR',
    'currency_choices' => ['EUR','USD']
));
````

That way we would be able to initialize it both with constructor (this is what extension would do) and with form factory (useful for more explicit tests).